test/system/src/test/java/io/pravega/test/system/MultiReaderWriterWithFailOverTest.java

```diff
@@ -0,0 +1,277 @@
+        List<URI> conUris = conService.getServiceDetails();
+        log.debug("Pravega Controller service  details: {}", conUris);
+
+        //4.start 2 instances of pravega segmentstore
+        Service segService = new PravegaSegmentStoreService("multisegmentstorewithfailover", zkUri, conUris.get(0));
+        if (!segService.isRunning()) {
+            segService.start(true);
+        }
+        segService.scaleService(2, true);
+        List<URI> segUris = segService.getServiceDetails();
+        log.debug("Pravega segmentstore service  details: {}", segUris);
+    }
+
+    @Before
+    public void setup() {
+
+        //1. Start 1 instance of zookeeper
+        Service zkService = new ZookeeperService("zookeeper");
+        if (!zkService.isRunning()) {
+            zkService.start(true);
+        }
+        List<URI> zkUris = zkService.getServiceDetails();
+        log.debug("Zookeeper service details: {}", zkUris);
+        //get the zk ip details and pass it to bk, host, controller
+        URI zkUri = zkUris.get(0);
+
+        // Verify controller is running.
+        controllerInstance = new PravegaControllerService("multicontrollerwithfailover", zkUri);
+        Assert.assertTrue(controllerInstance.isRunning());
+        List<URI> conURIs = controllerInstance.getServiceDetails();
+        log.info("Pravega Controller service instance details: {}", conURIs);
+
+        // Verify segment store is running.
+        segmentStoreInstance = new PravegaSegmentStoreService("multisegmentstorewithfailover", zkUri, conURIs.get(0));
+        Assert.assertTrue(segmentStoreInstance.isRunning());
+        log.info("Pravega segment store instance details: {}", segmentStoreInstance.getServiceDetails());
+    }
+
+    @After
+    public void tearDown() {
+        controllerInstance.stop();
+        segmentStoreInstance.stop();
+    }
+
+    @Test(timeout = 600000)
+    public void multiReaderWriterWithFailOverTest() throws Exception {
+
+        log.info("Test with 2 controller, SSS instances running and without a failover scenario");
+        readWriteTest();
+
+        //scale down SSS by 1 instance
+        segmentStoreInstance.scaleService(1, true);
+        Thread.sleep(60000);
+        log.info("Test with 1 SSS instance down");
+        readWriteTest();
+
+        //scale down controller by 1 instance + scale up SSS back to 2 instances
+        segmentStoreInstance.scaleService(2, true);
+        Thread.sleep(60000);
+        controllerInstance.scaleService(1, true);
+        Thread.sleep(60000);
+        log.info("Test with 1 controller instance down");
+        readWriteTest();
+
+        //scale down 1 instance of both controller, SSS
+        segmentStoreInstance.scaleService(1, true);
+        Thread.sleep(60000);
+        log.info("Test with 1 controller  and 1 SSS instance down");
+        readWriteTest();
+    }
+
+    private void readWriteTest() throws InterruptedException, ExecutionException {
```

Comment:
I don't see any cleanup being done here. The writers and readers should be closed and the stream/reader group should be deleted afterwards.